### ISSUE TITLE
Using Only SSH with Private Bitbucket Packages Failed with Composer v2.0.4

### ISSUE THREAD

**Issue by lloricode** [composer/composer#9400]
previous issue #9369

Our `composer.json`:
https://gist.github.com/lloricode/d965932ff7d6e4ea25bae5b8e52d65cc

Output `composer u` when using `v2.0.3` (works fine):
https://gist.github.com/lloricode/8a588f081f39631e36cdb1586decf578

Output `composer u` when using `v2.0.4` (not expected):
https://gist.github.com/lloricode/a89278cf37e7bdf1594a8cd07b5869d1

Output `composer i` when using `v2.0.3` (works fine): vendor remove before run `install`
https://gist.github.com/lloricode/194612bab18a186d81f245cbab5ed9b5

Output `composer i` when using `v2.0.4` (works fine): vendor remove before run `install`
https://gist.github.com/lloricode/7b896a1653ed880af13b0426b9156c90

notice that when using `composer u -n` it works fine with `composer v2.0.4`

**Comment by archon810** [comment]
Same issue here. Was very confused.

**Comment by archon810** [comment]
Reverting to 2.0.3 fixes the issue for us. No need for any bitbucket oauth token setup which 2.0.4 forced, and the urls are back to git@bitbucket.org and not https://bitbucket.org.

**Comment by kmbt** [comment]
Hi, I have managed to work around some issues with composer forcing the use of https instead of ssh by setting the repository type to "git" instead of "vcs". Perhaps it's worth a try.

**Comment by lloricode** [comment]
> Hi, I have managed to work around some issues with composer forcing the use of https instead of ssh by setting the repository type to "git" instead of "vcs". Perhaps it's worth a try.

Hmm, it seems make sense, will try this later

**Comment by lloricode** [comment]
> Hi, I have managed to work around some issues with composer forcing the use of https instead of ssh by setting the repository type to "git" instead of "vcs". Perhaps it's worth a try.

@kmbt, I  did not notice you said `https instead of ssh`, many of our team using ssh, so temporary fix is to use `composer v2.0.3` in local environment (when using `composer update`), while in remote we can use `composer v2.0.4` since it only work when using `composer install` command

**Comment by The-Hasanov** [comment]
temporary solution for me change type git to git-bitbucket
```
        {
            "type": "git-bitbucket",
            "name": "*****/*****",
            "url": "https://bitbucket.org/*****/*****.git"
        }
```

**Comment by Seldaek** [comment]
This is due to 985cef790f07c7b541badfcfd9913e6da7356244 which was well intentioned as I figured it was a good thing to make all bitbucket URLs use the API, but I guess not if people rely on this to force clone the repo. I'll revert the commit for 2.0.5. Sorry for the trouble.

**Comment by kmbt** [comment]
@lloricode 

> @kmbt, I did not notice you said `https instead of ssh`, many of our team using ssh, so temporary fix is to use `composer v2.0.3` in local environment (when using `composer update`), while in remote we can use `composer v2.0.4` since it only work when using `composer install` command

Oh, I can see now that my message was a bit ambiguous. Sorry for that. I have corrected it in order to make it clearer :)

**Comment by lloricode** [comment]
> @lloricode
> 
> > @kmbt, I did not notice you said `https instead of ssh`, many of our team using ssh, so temporary fix is to use `composer v2.0.3` in local environment (when using `composer update`), while in remote we can use `composer v2.0.4` since it only work when using `composer install` command
> 
> Oh, I can see now that my message was a bit ambiguous. Sorry for that. I have corrected it in order to make it clearer :)

I see, will try, but better to wait `composer 2.0.5`, while waiting, we just use `2.0.3` for a while, since there is nothing impact to our application, important is to install packages
